### ISSUE TITLE
Problems with SMB and letter casing

### ISSUE THREAD

**Issue by My1** [owncloud/core#17138]
I found a pretty weird error when trying to access an SMB-Share

main data:
OC8.0.3 on opensuse server
SMB-Share on Win-Server 2008r2
basically the path (in windows is \172.16.x.y\share$<username>)
so I set in owncloud I set 172.16.x.y as server-IP share$ as sharename and $user as path
the problem is we have a user named TLehrer and the smb-folder named tlehrer (and similar for all other users)
and I get the error: url_stat(): path 'TLehrer' not found at /opt/lampp/htdocs/owncloud/apps/files_external/3rdparty/smb4php/smb.php#260
due to some manual testing with the URL I found out that I can get into 
http://IP/owncloud/index.php/apps/files/?dir=/Personal/tlehrer
but not into
http://IP/owncloud/index.php/apps/files/?dir=/Personal/TLehrer
the latter one bouces me back to the root.
so in other words it tries to be case sensitive for smb which is a fact that shouldnt be on a Win Server, althou I do admit that that wasnt really best practice, but I dont really have control over the user and older names as the user names come from an LDAP, but I will talk to the higher-ups about that...
but still this is an error that should be rersolved (for example with a tick "this is a wondows server" similar to what we have in LDAP.

**Comment by LukasReschke** [comment]
Thanks a lot for reporting issues back to us!

We need some more information to properly support you with your issue!

Can I ask you to follow our guidelines for submitting bugs as described here: https://github.com/owncloud/core/blob/master/CONTRIBUTING.md

**Comment by My1** [comment]
### Steps to reproduce
1. have an SMB share from a WinServer mounted in owncloud
2. try to get to a path that exists but with different case (either through URL manipulation or with the $user variable)

### Expected behaviour
get on the folder because SMB/Windows doesnt care about the case

### Actual behaviour
it does care about the case and thrws an error/bounces me back.

### Server configuration
**Operating system**:
openSUSE 13.2

**Web server:**
XAMPP for Linux

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
8.0.3

**Updated from an older ownCloud or fresh install:**
fresh

**List of activated apps:**
file lock, ext storage, ldap user/group backend, updater

**The content of config/config.php:**

```
Insert your config.php content here
(Without the database password, passwordsalt and secret)
<?php
$CONFIG = array (
  'instanceid' => 'oc2pfl40i5q5',
  'passwordsalt' => 'HIDDEN',
  'secret' => 'HIDDEN',
  'trusted_domains' => 
  array (
    0 => 'localhost',
    1 => '192.168.69.2',
  ),
  'datadirectory' => '/opt/lampp/htdocs/owncloud/data',
  'overwrite.cli.url' => 'http://localhost/owncloud',
  'dbtype' => 'mysql',
  'version' => '8.0.3.4',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => 'HIDDEN',
  'installed' => true,
  'ldapIgnoreNamingRules' => false,
  'loglevel' => 0,
);
```

**Are you using external storage, if yes which one:** smb

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** ActiveDirectory over LDAP

#### LDAP configuration (delete this part if not used)

```
I did it on the mysqlserver since I dont use sqlite, I hope that's okay...

ppid 	configkey 	configvalue 	
user_ldap 	cleanUpJobOffset 	0
user_ldap 	enabled 	yes
user_ldap 	installed_version 	0.5.0
user_ldap 	s01has_memberof_filter_support 	1
user_ldap 	s01home_folder_naming_rule 	
user_ldap 	s01last_jpegPhoto_lookup 	0
user_ldap 	s01ldap_agent_password 	XXXXXXXXX
user_ldap 	s01ldap_attributes_for_group_search 	
user_ldap 	s01ldap_attributes_for_user_search 	
user_ldap 	s01ldap_backup_host 	
user_ldap 	s01ldap_backup_port 	
user_ldap 	s01ldap_base 	DC=Domain,DC=de
user_ldap 	s01ldap_base_groups 	DC=Domain,DC=de
user_ldap 	s01ldap_base_users 	DC=Domain,DC=de
user_ldap 	s01ldap_cache_ttl 	600
user_ldap 	s01ldap_configuration_active 	1
user_ldap 	s01ldap_display_name 	displayname
user_ldap 	s01ldap_dn 	CN=Admin User,OU=Operatoren,OU=Benutzer,DC=Domain,DC=de
user_ldap 	s01ldap_email_attr 	mail
user_ldap 	s01ldap_experienced_admin 	0
user_ldap 	s01ldap_expert_username_attr 	sAMAccountName
user_ldap 	s01ldap_expert_uuid_group_attr 	
user_ldap 	s01ldap_expert_uuid_user_attr 	
user_ldap 	s01ldap_group_display_name 	cn
user_ldap 	s01ldap_group_filter 	(&(|(objectclass=group)(objectclass=top))(|(cn=Group A)(cn=Group B)))
```

### Client configuration
**Browser:** FF 38.0.5

**Operating system:** Win7x64

### Logs
#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
(as if I could paste 14,6MB of text here, so here's the main point:
Debug	core	OC\Tags::__construct, tags: Array ( ) 	2015-06-16T05:39:37+00:00
Debug	OC\Files\Cache\Scanner	!!! Path '' is not accessible or present !!!	2015-06-16T05:39:37+00:00
Error	PHP	url_stat(): path 'TLehrer' not found at /opt/lampp/htdocs/owncloud/apps/files_external/3rdparty/smb4php/smb.php#260	2015-06-16T05:39:37+00:00
```

**Comment by LukasReschke** [comment]
Mhm. Not sure how we handle / should handle case sensitivity in this case since it might be expected behaviour due to the file cache. @icewind1991 any idea?

**Comment by My1** [comment]
is there some way to drop the cache since it isnt really needed in our use-case since we use owncloud mainly as smb/webdav bridge

**Comment by PVince81** [comment]
The cache (which is actually more an index than a cache) is required for OC to map file metadata with files (ex: shares, favorite, etc). It is not possible to disable it.

Regarding file casing, this is a known issue, there have been discussions here about how to solve it in the future here: https://github.com/owncloud/client/issues/1348 and https://github.com/owncloud/core/issues/17161 (but nothing like a quickfix, needs some architectural changes)

**Comment by My1** [comment]
well we need neither shares nor favorites (actually I disabled the share app, but the fact that this is hard to solve is not very nice but it isnt rly your fault.

**Comment by PVince81** [comment]
@jvillafanez

**Comment by My1** [comment]
I must say that I am really happy that this bug isnt left in the desert.

**Comment by jvillafanez** [comment]
Works for me:
* Ubuntu 14.04
* PHP 5.5.9
* smbclient Version 4.1.6-Ubuntu
* php5-libsmbclient installed
* ownCloud 9.1.1
* SMB share in Windows Server 2008 RC2

<img width="561" alt="screen shot 2016-09-22 at 10 50 34" src="https://cloud.githubusercontent.com/assets/1477829/18742044/89a3be80-80b2-11e6-9c33-655fe8f1f727.png">


@My1 Could you try with newer versions?

**Comment by My1** [comment]
well I will have to set everything up again (dont use oc atm) but I can.

**Comment by jvillafanez** [comment]
You might want to download an ownCloud appliance just to test.

**Comment by My1** [comment]
nice idea. thanks. I'll try when home.

**Comment by PVince81** [comment]
Closing due to lack of feedback.
